include/sys/queued_operation.h

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_QUEUED_OPERATION_H_
+#define ZEPHYR_INCLUDE_SYS_QUEUED_OPERATION_H_
+
+#include <kernel.h>
+#include <zephyr/types.h>
+#include <sys/async_notify.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Forward declaration */
+struct queued_operation;
+struct queued_operation_manager;
+
+/**
+ * @defgroup resource_mgmt_queued_operation_apis Queued Operation APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/** @internal */
+#define QUEUED_OPERATION_PRIORITY_POS ASYNC_NOTIFY_EXTENSION_POS
+/** @internal */
+#define QUEUED_OPERATION_PRIORITY_BITS 8U
+/** @internal */
+#define QUEUED_OPERATION_PRIORITY_MASK BIT_MASK(QUEUED_OPERATION_PRIORITY_BITS)
+
+/**
+ * @brief Special priority value to indicate operation should be
+ * placed last in current queue.
+ *
+ * This is equvalent to providing the lowest priority but uses a
+ * constant-time insertion.
+ */
+#define QUEUED_OPERATION_PRIORITY_APPEND ((int)QUEUED_OPERATION_PRIORITY_MASK + 1)
+
+/**
+ * @brief Special priority value to indicate operation should be
+ * placed first in the current queue.
+ *
+ * This is similar to providing the highest priority but uses a
```

Comment:
As a first-time reader, the fact that this uses "similar" but the previous docstring uses "equivalent" makes me wonder if this is somehow not "equivalent" -- if not, why not? If it is equivalent, why not just use "equivalent" again?

Reply:
The standard insertion uses FIFO within a priority.  When you append the effect is FIFO, which is equivalent.  When you prepend the effect is LIFO, which is not equivalent.

I assumed going into the details would be distracting, as the difference is implicit in the description.  I can change "equivalent/similar to" to "like" for both if that reads better.